services/time/src/lib.rs

```diff
@@ -111,62 +113,100 @@ transactions! {
     }
 }
 
-impl Transaction for TxTime {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.pub_key())
-    }
+#[derive(Debug, Fail)]
+#[repr(u8)]
+enum Error {
+    #[fail(display = "Not authored by a validator")]
+    UnknownSender = 0,
 
-    fn execute(&self, view: &mut Fork) -> ExecutionResult {
-        let validator_keys = Schema::new(&view).actual_configuration().validator_keys;
+    #[fail(display = "The validator time is longer than the proposed one")]
+    ValidatorTimeIsLonger = 1,
+}
 
-        // The transaction must be signed by the validator.
-        let signed = validator_keys.iter().any(|&validator| {
-            validator.service_key == *self.pub_key()
-        });
+impl From<Error> for ExecutionError {
+    fn from(value: Error) -> ExecutionError {
+        ExecutionError::new(value as u8)
+    }
+}
+
+impl TxTime {
+    fn signed_by_validator(&self, snapshot: &Snapshot, key: &PublicKey) -> ExecutionResult {
```

Comment:
- Cannot `key` be taken from `self`?
- I don't quite like the method name; maybe, `check_signed_by_validator` could be better?